runtime/jcl/common/mgmtinit.c

```diff
@@ -944,17 +944,39 @@ getArrayIndexFromManagerID(J9JavaLangManagementData *mgmt, UDATA id)
 }
 
 static void
-getSegmentSizes(J9JavaVM *javaVM, J9MemorySegmentList *segList, U_64 *storedSize, U_64 *storedUsed, U_64 *storedPeakSize, U_64 *storedPeakUsed)
+getSegmentSizes(J9JavaVM *javaVM, J9MemorySegmentList *segList, U_64 *storedSize, U_64 *storedUsed, U_64 *storedPeakSize, U_64 *storedPeakUsed, BOOLEAN isCodeCacheSegment)
 {
 	U_64 used = 0;
 	U_64 committed = 0;
+	UDATA *mccCodeCache;
+	UDATA *warmAlloc;
+	UDATA *coldAlloc;
 	J9JavaLangManagementData *mgmt = javaVM->managementData;
 
 	omrthread_monitor_enter(segList->segmentMutex);
 
 	MEMORY_SEGMENT_LIST_DO(segList, seg)
+	if (isCodeCacheSegment) {
+		/* The JIT code cache grows from both ends of the segment: the warmAlloc pointer upwards from the start of the segment
+		 * and the coldAlloc pointer downwards from the end of the segment. The free space in a JIT code cache segment is the
+		 * space between the warmAlloc and coldAlloc pointers. See jit/codert/MultiCodeCache.hpp, the contract with the JVM is
+		 * that the address of the TR_MCCCodeCache structure is stored at the beginning of the segment, and that the warmAlloc
+		 * and coldAlloc pointers are the first two fields in TR_MCCCodeCache.
+		 */
+		mccCodeCache = *((UDATA**)seg->heapBase);
```

Comment:
Is there a structure we can use to access these fields?  and the warm/cold alloc pointers?

Reply:
the code was duplicated from javadump, not sure if there is a structure we can use.

Reply:
No, there isn't mainly because VM code typically does not include JIT header files. We could create a struct in one of the VM header files (with just those 2 fields) and embed this struct in the CodeCache at the very beginning.